content/docs/visualization/template_reference.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Template reference
+sort_rank: 6
+---
+
+# Template reference
+
+Prometheus has support for templating in both the summary/description field of
+alerts and also exposed as html endpoints. Templates have the ability to run
+queries against the databases, iterate over data, use conditionals, format data
+etc. This is based on the [go templating](http://golang.org/pkg/text/template/)
+system.
+
+## Data Structures
+
+The primary structure is the sample, defined as:
```

Comment:
"Any Prometheus query run via the template language returns a slice of samples. A sample is defined as:"